tests/dragonfly/replication_test.py

```diff
@@ -226,3 +226,57 @@ async def disconnect(replica, c_replica, crash_type):
 
     # Check master survived all disconnects
     assert await c_master.ping()
+
+
+"""
+Test crashing master and letting replicas re-connect to it.
+"""
+
+master_crash_cases = [
+    (8, [8], 100),
+    (6, [6, 6, 6], 500),
+    (4, [2] * 8, 500),
+]
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("t_master, t_replicas, n_keys", master_crash_cases)
+async def test_master_crash(df_local_factory, t_master, t_replicas, n_keys):
+    master = df_local_factory.create(port=1111, proactor_threads=t_master)
+    replicas = [
+        df_local_factory.create(port=BASE_PORT+i+1, proactor_threads=t)
+        for i, t in enumerate(t_replicas)
+    ]
+
+    master.start()
+    for replica in replicas:
+        replica.start()
+
+    c_master = aioredis.Redis(port=master.port)
+    c_replicas = [aioredis.Redis(port=replica.port) for replica in replicas]
+
+    # Fill master with test data
+    await batch_fill_data_async(c_master, gen_test_data(n_keys, seed=0))
+
+    # Do full sync
+    async def full_sync(c_replica):
+        await c_replica.execute_command("REPLICAOF localhost " + str(master.port))
+        await wait_available_async(c_replica)
+
+    await asyncio.gather(*(full_sync(c) for c in c_replicas))
+
+    # Crash master
+    master.stop(kill=True)
```

Comment:
I'm planning to make this test more robust, where we crash the master in different phases and make sure the replicas recover (unblock) and re-connnect once it becomes available